internal/provider/machine_pool_data_source.go

```diff
@@ -156,6 +161,9 @@ func (d *MachinePoolDataSource) Read(ctx context.Context, req datasource.ReadReq
 	}
 
 	data.ID = types.Int64Value(result.MachinePool.ID)
+	if result.MachinePool.NetworkProfileId != nil {
```

Comment:
do we need to set `data.NetworkProfileID` back to `nil` if `result.MachinePool.NetworkProfileId` is set to nil?